### PR TITLE
PR for the micro editor in the conda feedstock

### PR DESCRIPTION
**Pull request by abhi18av**
Hi conda-forge team

This is a PR for the addition of a cross platform terminal based text editor written in go called micro

https://github.com/zyedidia/micro

### CODE REVIEW

File: recipes/micro/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# Note: there are many handy hints in comments in this example -- remove them when you've finalized your recipe
```
Comment (synapticarbors) [inline recipes/micro/meta.yaml]: Please delete all comments

File: recipes/micro/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+ #   - setuptools
+    # if your project compiles code (such as a C extension) then add `toolchain` as a build requirement.
+ # run:
+  #  - python
+
+test:
+  # Some package might need a `test/commands` key to check CLI.
+  # List all the packages/modules that `run_test.py` imports.
+  commands:
+    - python --version
```
Comment (synapticarbors) [inline recipes/micro/meta.yaml]: You need to test your actual package. https://conda.io/docs/user-guide/tasks/build-packages/define-metadata.html#test-section

File: recipes/micro/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+# You may need the openssl package, available on conda-forge
+#  `conda install openssl -c conda-forge``
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+#  fn: {{ name }}-{{ version }}.tar.g
+  url: https://github.com/zyedidia/micro/releases/download/v1.3.3/micro-{{ version }}-linux64.tar.gz
```
Comment (synapticarbors) [inline recipes/micro/meta.yaml]: It looks like you need to use selectors and specify a different tarball for each platform:

https://github.com/zyedidia/micro/releases

https://conda.io/docs/user-guide/tasks/build-packages/define-metadata.html#preprocessing-selectors

Or perhaps you should be building from source.

File: recipes/micro/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  #  - python
+
+test:
+  # Some package might need a `test/commands` key to check CLI.
+  # List all the packages/modules that `run_test.py` imports.
+  commands:
+    - python --version
+
+about:
+  home: http://github.com/simplejson/simplejson
```
Comment (synapticarbors) [inline recipes/micro/meta.yaml]: This `home` url is not correct.

**Comment by abhi18av** [comment]
@synapticarbors thanks for the review and suggestions ;)

I'll update the PR as soon as I'm near my machine again. Looking forward to having the Micro editor in the channel soon.

File: recipes/micro/meta.yaml (hunk 1 above)
Comment (abhi18av) [reply recipes/micro/meta.yaml]: Done.

File: recipes/micro/meta.yaml (hunk 4 above)
Comment (abhi18av) [reply recipes/micro/meta.yaml]: Done

File: recipes/micro/meta.yaml (hunk 3 above)
Comment (abhi18av) [reply recipes/micro/meta.yaml]: Done

File: recipes/micro/meta.yaml (hunk 2 above)
Comment (abhi18av) [reply recipes/micro/meta.yaml]: Hi @synapticarbors 

Now with this commit https://github.com/conda-forge/staged-recipes/pull/4291/commits/792e5d79a598b2e0036714d804786260356f1b9b  only the tests remains to be fixed for the package to be successfully done on CircleCI

Here's the CircleCI build log 
https://circleci.com/gh/conda-forge/staged-recipes/21129#build-parameters/containers/0

**Comment by abhi18av** [comment]
@synapticarbors Hi :)

Could you help me with this error message 
```sh

subprocess.CalledProcessError: Command '['/bin/bash', '-x', '-e', '/staged-recipes/build_artefacts/micro_1510132807744/work/micro-1.3.3/conda_build.sh']' returned non-zero exit status 1.

```

Full log can be read here 
https://circleci.com/gh/conda-forge/staged-recipes/21308?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by abhi18av** [comment]
Damn, I am really surprised I had so many open PRs from 2017!

Apologies for this 🗻

**Pull request closed without merging**